modules/host/negotiatesettings.go

```diff
@@ -26,54 +26,60 @@ func (h *Host) capacity() (total, remaining uint64, err error) {
 	return total, remaining, nil
 }
 
+// externalSettings compiles and returns the external settings for the host.
+func (h *Host) externalSettings() (modules.HostExternalSettings, error) {
+	totalStorage, remainingStorage, err := h.capacity()
+	if err != nil {
+		return modules.HostExternalSettings{}, err
+	}
```

Comment:
symlinks do seem a little strange. If I make a folder `/bar` and tell the host to track it, I would expect the host to have some internal state that contains `/bar`. As for browsing, I'm skeptical that there's much utility to being able to navigate to all your storage folders from one place. Why would you want to browse your storage folders anyway? And even if I did want to browse them, my first thought would be to run `siac host` and then navigate to the listed folders, not `cd` to my Sia/host directory and follow a bunch of symlinks.

The latter scenario is a great example of where a generic `Either` type would be useful, i.e. `Either<StorageFolder, error>`. Then you could return a slice of those. We could simulate it by making a new type, or return a slice of pointers (`nil` meaning an unspecified error occurred, though it would be logged), or return a slice of errors, or return a single error alongside the valid entries. I think my preference would be the latter, using a custom error type so that the valid entries could be recovered:
```go
sfs, err := host.StorageFolders()
if sfe, ok := err.(*host.StorageFolderError); ok {
    for _, f := range sfe.ErrFolders {
        fmt.Println("Warning: could not parse folder", f)
    }
} else if err != nil {
    return err
}
// this path will still be reached if the error is a StorageFolderError
for _, f := range sfe.Folders {
    fmt.Println(f)
}
```